File: zero-server/api/api-auth/src/main/java/run/ice/zero/api/auth/model/oauth2/OAuth2.java
package run.ice.zero.api.auth.model.oauth2;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import run.ice.zero.common.model.Serializer;

/**
 * @author DaoDao
 */
@Schema(title = "OAuth2", description = "OAuth 2.1 协议 : https://datatracker.ietf.org/doc/draft-ietf-oauth-v2-1/")
@Data
public class OAuth2 implements Serializer {

    @Schema(title = "access_token", description = "访问令牌", example = "xxx")
    private String access_token;

    @Schema(title = "refresh_token", description = "刷新令牌", example = "yyy")
    private String refresh_token;

    @Schema(title = "id_token", description = "ID 令牌", example = "zzz")
    private String id_token;

    @Schema(title = "scope", description = "授权范围", example = "openid")
    private String scope;

    @Schema(title = "token_type", description = "令牌类型", example = "bearer")
    private String token_type;

    @Schema(title = "expires_in", description = "过期时间", example = "123")
    private Integer expires_in;

}

File: admin/src/api/auth/oauth2.ts
import { http } from "@/utils/http";
import type { No, Ok, Request, Response } from "@/api";

/** 登录 传入 username & password 信息，获取 token */
export async function login(
  request: Request<Login>
): Promise<Response<OAuth2>> {
  return http.post<Request<Login>, Response<OAuth2>>(`/auth/api/login`, {
    data: request
  });
}

/** 退出 header 中传入 token 信息，退出登录 */
export async function logout(request: Request<No>): Promise<Response<Ok>> {
  return http.post<Request<No>, Response<Ok>>(`/auth/api/logout`, {
    data: request
  });
}

export type Login = {
  /** username */
  username: string;
  /** password */
  password: string;
  /** captchaId 验证码 ID */
  captchaId: string;
  /** captchaCode 验证码 */
  captchaCode: string;
};

export type OAuth2 = {
  /** access_token */
  access_token: string;
  /** refresh_token */
  refresh_token: string;
  /** id_token */
  id_token: string;
  /** scope */
  scope: string;
  /** token_type */
  token_type: string;
  /** expires_in */
  expires_in: number;
};
